sapi/fpm/config.m4

```diff
@@ -555,18 +555,39 @@ if test "$PHP_FPM" != "no"; then
   fi
 
   if test "$PHP_FPM_ACL" != "no" ; then
+    AC_MSG_CHECKING([for acl user/group permissions support])
     AC_CHECK_HEADERS([sys/acl.h])
-    dnl *BSD has acl_* built into libc.
-    AC_CHECK_FUNC(acl_free, [
-      AC_DEFINE(HAVE_FPM_ACL, 1, [ POSIX Access Control List ])
-    ],[
-      AC_CHECK_LIB(acl, acl_free, [
-        PHP_ADD_LIBRARY(acl)
-        AC_DEFINE(HAVE_FPM_ACL, 1, [ POSIX Access Control List ])
-      ],[
-        AC_MSG_ERROR(libacl required not found)
-      ])
-    ])
+
+    AC_COMPILE_IFELSE([AC_LANG_SOURCE([[#include <sys/acl.h>
+      int main()
+      {
+        acl_t acl;
+        acl_entry_t user, group;
+        acl = acl_init(1);
+        acl_create_entry(&acl, &user);
+        acl_set_tag_type(user, ACL_USER);
+        acl_create_entry(&acl, &group);
+        acl_set_tag_type(user, ACL_GROUP);
+        acl_free(acl);
+        return 0;
+      }
+    ]])], [
+      AC_CHECK_LIB(acl, acl_free, 
+        [PHP_ADD_LIBRARY(acl)
+          have_fpm_acl=yes
+          AC_MSG_RESULT([yes])
+        ],[
+      	  have_fpm_acl=no
+      	  AC_MSG_RESULT([no])
+        ])
+    ], [
```

Comment:
It's in the wrong branch as this should be in the `action-if-false` and not `action-if-true`. Also if you think that just checking `acl_free` is not enough, maybe you should then do a similar check with the use of `AC_LINK_IFELSE`...

Reply:
Because I did not consider it as fallback check but if the code does compile then check the lib.

Reply:
But how can this compile without the lib on Linux? From the comment it's available without linking on OpenBSD (and maybe other BSD based systems) but for Linux you need to link it with libacl. It means you would need `AC_LINK_IFELSE` for such check IMHO.

Reply:
that s the difference b/w `AC_RUN_IFELSE` and `AC_COMPILE_IFELSE`, the latter is sufficient just to compile without linking it. but I did not think of the openbsd's case tough.

Reply:
Ah I got it now - yeah you are right for compiling it just needs headers.